docker/configure_workers_and_start.py

```diff
@@ -446,11 +509,11 @@ def generate_worker_files(
 
         # Check if more than one instance of this worker type has been specified
         worker_type_total_count = worker_types.count(worker_type)
-        if worker_type_total_count > 1:
```

Comment:
What was the logic behind removing the gate? I saw your comment about unsupported types passing through, but why not check so we don't end up running code unnecessarily?

Reply:
I had more info on that somewhere I had left as a comment. I'll have to find it.

Reply:
Found it in PR [#14153 ](https://github.com/matrix-org/synapse/pull/14153#issuecomment-1279936319). But I wasn't able to get clarification on it.

Reply:
> What was the logic behind removing the gate? I saw your comment about unsupported types passing through, but why not check so we don't end up running code unnecessarily?

I guess to be more explicit? Plus, this allows repurposing the function to handle the stream writers without a bunch of extra work. Python isn't my native language, and I'm not totally comfortable with it yet. Removing this gating allowed the function to run through and process everything worker related and adds less than a second(give or take) to the running of a bootstrap script.

Reply:
> I welcome suggestions on what to rename `add_sharding_to_shared_config()` since it does a little bit more now(and probably should have been all along). Perhaps `add_streamwriters_and_sharding_to_shared_config()`? Feels kinda long, for a function name. OTOH, have you seen Windows function names? `Win64DoThisAndThatAndThoseOthersAndThenResetAllSettingsToDefaultsWithoutAsking_45927()`
> 
> @anoadragon453 Since you wrote the original function, I feel like yours is the input for this.

Agreed. `handle_worker_configuration()` felt kind of generic...

Reply:
Renamed the function name to something ridiculously long for the moment. I'll refactor this into two functions later and fix it then.